op-node/rollup/derive/engine_controller.go

```diff
@@ -388,13 +415,25 @@ func (e *EngineController) InsertUnsafePayload(ctx context.Context, envelope *et
 			FinalizedBlockHash: e.finalizedHead.Hash,
 		}
 
-		fcuRes, err := e.engine.ForkchoiceUpdate(ctx, &fcuReq, nil)
-		if fcuRes.PayloadStatus.Status == eth.ExecutionValid {
-			log.Info("engine processed data successfully")
-			e.needFCUCall = false
-			return nil
-		} else {
-			return NewTemporaryError(fmt.Errorf("engine failed to process inconsistent data: %w", err))
+		for attempts := 0; attempts < maxFCURetryAttempts; attempts++ {
```

Comment:
In this entire section of code within the else block, I understand that it is all about attempting to instruct op-geth to repair the data through FCU when a data inconsistency is detected. I think we should abstract this into a method with an appropriate name to make the code easier to understand.

Reply:
updated